config/m4/compiler.m4

```diff
@@ -4,9 +4,9 @@
 #
 
 #
-# Initialize CFLAGS
+# Initialize internal CFLAGS
 #
-CFLAGS="-g -Wall -Werror $UCX_CFLAGS"
+AM_CFLAGS="-g -Wall -Werror $UCX_CFLAGS"
```

Comment:
Kind of? Look into this the black magic just got deeper. I'm going to change this to the previous 'bad version' that still brings in user CFLAGS. Something more comprehensive will be needed to do things 'the right way'.